src/unix/solarish/illumos.rs

```diff
@@ -68,6 +68,8 @@ pub const B4000000: ::speed_t = 31;
 // sys/systeminfo.h
 pub const SI_ADDRESS_WIDTH: ::c_int = 520;
 
+pub const PTHREAD_MUTEX_DEFAULT: ::c_int = 8;
```

Comment:
This define cannot be changed in libc so soon.  It was only changed in illumos in February 2024.  Any running systems which predate the change (which there are many) would potentially malfunction if newly-compiled rust binaries use this newer value.

Reply:
so we are ok if unit tests do not necessarily pass then ?